sql/core_clients_daily_v1.sql

```diff
@@ -0,0 +1,77 @@
+WITH
+  numbered_duplicates AS (
+  SELECT
+    * REPLACE(LOWER(client_id) AS client_id),
+    ROW_NUMBER() OVER (PARTITION BY client_id, submission_date_s3, metadata.document_id ORDER BY metadata.timestamp DESC) AS n
+  FROM
+    telemetry_core_parquet_v3
+  WHERE
+    client_id IS NOT NULL ),
+  -- Deduplicating on document_id is necessary to get valid SUM values.
```

Comment:
❤️

Reply:
this is an excellent pattern, and I think we should use it everywhere, and maybe provide it as a non-materialized view

Reply:
In the future, when we're working with the data ingested directly to BQ, should we consider having a query upstream in the DAG that runs a `DELETE` of duplicates in the base table?

Reply:
> In the future, when we're working with the data ingested directly to BQ, should we consider having a query upstream in the DAG that runs a DELETE of duplicates in the base table?

Yes, I think this is something we should continue to consider and at some point do a focused test to make sure we have that logic 100% correct. For the time being, though, this non-destructive dedupe within the query is a safe and effective alternative.